packages/flutter/lib/src/material/dialog.dart

```diff
@@ -282,11 +285,35 @@ class AlertDialog extends StatelessWidget {
     this.shape,
     this.alignment,
     this.scrollable = false,
-  }) : assert(contentPadding != null),
-       assert(clipBehavior != null);
+  }) : assert(clipBehavior != null);
+
+  /// An optional icon to display at the top of the dialog.
+  ///
+  /// Typically, an [Icon] widget. Providing an icon centers the [title]'s text.
+  final Widget? icon;
+
+  /// Color for the [Icon] in the [icon] of this [AlertDialog].
+  ///
+  /// If null, [DialogTheme.iconColor] is used. If that is null, defaults to
+  /// color scheme's [ColorScheme.secondary] if [ThemeData.useMaterial3] is
+  /// true, black otherwise.
+  final Color? iconColor;
+
+  /// Padding around the [icon].
+  ///
+  /// If there is no [icon], no padding will be provided. Otherwise, this
+  /// padding is used.
+  ///
+  /// This property defaults to providing 24 pixels on the top, left, and right
+  /// of the [icon]. If [title] is _not_ null, an extra 16 pixels of bottom
+  /// padding is added to separate the [icon] from the [title]. If the [title]
+  /// is null and [content] is _not_ null, then no bottom padding is provided
+  /// (but see [contentPadding]). In any other case an extra 24 pixels of bottom
+  /// padding is added.
+  final EdgeInsetsGeometry? iconPadding;
```

Comment:
I don't think we need icon padding at all, since `icon` is a widget and can provide it as needed. Nevertheless, there should still be default top and bottom padding for the case where someone just provides an `Icon`